File: app/src/main/java/com/warunya/ricenearby/constant/Filter.java
package com.warunya.ricenearby.constant;

import java.util.HashMap;

public enum Filter {

    OneWeek(7),
    TwoWeeks(14),
    OneMonth(30),
    TwoMonths(60),
    ThreeMonths(90),
    SixMonths(180),
    All(0);

    int period;

    Filter(int period) {
        this.period = period;
    }

    public int getPeriod() {
        return period;
    }

    public static Filter parse(int type) {
        HashMap<Integer, Filter> creatorMap = new HashMap<>();
        creatorMap.put(OneWeek.ordinal(), OneWeek);
        creatorMap.put(TwoWeeks.ordinal(), TwoWeeks);
        creatorMap.put(OneMonth.ordinal(), OneMonth);
        creatorMap.put(TwoMonths.ordinal(), TwoMonths);
        creatorMap.put(ThreeMonths.ordinal(), ThreeMonths);
        creatorMap.put(SixMonths.ordinal(), SixMonths);
        creatorMap.put(All.ordinal(), All);

        return (creatorMap.get(type) == null) ? All : creatorMap.get(type);
    }
}

File: app/src/main/java/com/warunya/ricenearby/constant/Filter.kt
package com.warunya.ricenearby.constant

enum class Filter(val period: Int) {

    OneWeek(7),
    TwoWeeks(14),
    OneMonth(30),
    TwoMonths(60),
    ThreeMonths(90),
    SixMonths(180),
    All(0);

    companion object {
        fun parse(type: Int): Filter {
            val creatorMap = HashMap<Int, Filter>()
            creatorMap[OneWeek.ordinal] = OneWeek
            creatorMap[TwoWeeks.ordinal] = TwoWeeks
            creatorMap[OneMonth.ordinal] = OneMonth
            creatorMap[TwoMonths.ordinal] = TwoMonths
            creatorMap[ThreeMonths.ordinal] = ThreeMonths
            creatorMap[SixMonths.ordinal] = SixMonths
            creatorMap[All.ordinal] = All

            return if (creatorMap[type] == null) All else creatorMap[type]!!
        }
    }
}